docs/src/guide/hello_world.md

```diff
@@ -0,0 +1,290 @@
+
+The `update` function is the heart of the app. It responds to events by (optionally) updating the state and requesting some effects by returning `Command`s. The `Command` type bundles an effect with a message that the app should be sent when the effect is complete, but unfortunately for us, the `Render` effect does not return anything, which makes it a bad example to demonstrate the distinction between `Command` and `Effect`. We'll see a better one shortly.
+
+All our `update` function does is ignore all its arguments and ask the Shell to render the screen.
+
+Finally, the `view` function returns the representation of what we want the Shell to show on screen. To start with, it's a simple hello world.
+
+That's a working hello world done, lets try it. For now we'll do it from tests. That's not really a limitation, in fact, this is the intended way for apps to be developed with Crux - from inside out, with unit tests, focusing on behavior first and presentation later.
+
+```rust
+#[cfg(test)]
+mod tests {
+    use crux_core::{Core, Command};
+    use super::*;
+
+    #[test]
+    fn hello_says_hello_world() {
+        let capabilities = Capabilities::default();
+        let hello = Hello::default();
+        let model = ();
+
+        // Call 'update' and receive commands
+        let commands = hello.update((), &mut model, &capabilities);
+
+        let Command { effect: actual_effect, .. } = commands[0];
+        let expected_effect = Effect::Render;
+
+        // Check update asked us to `Render`
+        assert_eq!(actual_effect, expected_effect);
+
+        let actual_view = hello.view(&model);
+        let expected_view = "Hello World".to_string();
+
+        assert_eq!(actual_view, expected_view);
+    }
+}
+```
+
+It is a fairly underwhelming test, but it should pass (check with `cargo test`).
+
+## Counting up and down
+
+To make things more interesting, we'll add some behaviour. We'll teach the app to count up and down. First, we'll need a model, which represents the state. We could just use a number, but we'll use a struct instead, so that we can easily add more state later.
+
+```rust
+#[derive(Default)]
+struct Model {
+    count: isize,
+}
+```
+
+We need `Default` implemented to define the initial state. For now we derive it, as our state is quite simple. We also update the app to use the model:
+
+```rust
+impl<E, C> App<E, C> for Hello where
+    E: Clone + Serialize,
+    C: Capabilities<Render<E>>
+{
+// ...
+
+    type Model = Model;
+
+// ...
+
+    fn view(&self, model: &Model) -> ViewModel {
+        format!("Count is: {}", model.count)
+    }
+}
+```
+
+Great. All that's left is adding the behaviour. That's where `Event` comes in:
+
+```rust
+#[derive(Serialize, Deserialize)]
+enum Event {
+    Increment,
+    Decrement,
+    Reset,
+}
+```
```

Comment:
Suspect we might need an empty event type at the very start of this so the `Effect` macro doesn't go mad